internal/storageboxsubaccount/data_source_list_test.go

```diff
@@ -0,0 +1,127 @@
+package storageboxsubaccount_test
+
+import (
+	"fmt"
+	"slices"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+	"github.com/hashicorp/terraform-plugin-testing/statecheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfjsonpath"
+
+	"github.com/hetznercloud/hcloud-go/v2/hcloud/exp/kit/randutil"
+	"github.com/hetznercloud/hcloud-go/v2/hcloud/exp/labelutil"
+	"github.com/hetznercloud/hcloud-go/v2/hcloud/schema"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/storagebox"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/storageboxsubaccount"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/teste2e"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/testsupport"
+	"github.com/hetznercloud/terraform-provider-hcloud/internal/testtemplate"
+)
+
+func TestAccStorageBoxDataSourceList(t *testing.T) {
+	tmplMan := testtemplate.Manager{}
+
+	resStorageBox := &storagebox.RData{
+		StorageBox: schema.StorageBox{
+			Name:           fmt.Sprintf("subaccount-ds-list-%s", randutil.GenerateID()),
+			StorageBoxType: schema.StorageBoxType{Name: teste2e.TestStorageBoxType},
+			Location:       schema.Location{Name: teste2e.TestLocationName},
+		},
+		Password: storagebox.GeneratePassword(t),
+	}
+	resStorageBox.SetRName("default")
+
+	res1 := &storageboxsubaccount.RData{
+		StorageBox:    resStorageBox.TFID() + ".id",
+		HomeDirectory: "test-1",
+		Password:      storagebox.GeneratePassword(t),
+		Description:   "tf-e2e-subaccount-ds-1",
+		Labels: map[string]string{
+			"key": randutil.GenerateID(),
+		},
+		Raw: `
+			access_settings = {
+				reachable_externally = true
+				samba_enabled = false
+				ssh_enabled = true
+				webdav_enabled = false
+				readonly = true
+			}`,
+	}
+	res1.SetRName("default1")
+
+	res2 := &storageboxsubaccount.RData{
+		StorageBox:    resStorageBox.TFID() + ".id",
+		HomeDirectory: "test-2",
+		Password:      storagebox.GeneratePassword(t),
+		Description:   "tf-e2e-subaccount-ds-2",
+		Labels: map[string]string{
+			"key": randutil.GenerateID(),
+		},
+		Raw: fmt.Sprintf(`
+			access_settings = {
+				reachable_externally = false
+				samba_enabled = true
+				ssh_enabled = false
+				webdav_enabled = true
+				readonly = false
+			}
+
+			depends_on = [%s]`, res1.TFID()), // Only one subaccount may be created at the same time
```

Comment:
For snapshot I would understand, but for subaccounts, I think we should handle the error in the provider.

Or is this a left over from the copy pasta from snapshots?